llvm/tools/repo-fragments/RepoFragments.cpp

```diff
@@ -0,0 +1,124 @@
+//===- repo-fragments - Dump the names and digests using a ticket file. --===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/MC/MCRepoTicketFile.h"
+#include "llvm/Support/CommandLine.h"
+
+#include "pstore/dump/mcrepo_value.hpp"
+
+using namespace llvm;
+
+namespace {
+
+static cl::OptionCategory ToolOptions("Tool Options");
+
+static cl::opt<std::string> RepoPath("repo", cl::Optional,
+                                     cl::desc("Path of the program repository"),
+                                     cl::init("./clang.db"),
+                                     cl::cat(ToolOptions));
+static cl::opt<std::string> TicketPath(cl::Positional,
+                                       cl::desc("<Path of the ticket file>"),
+                                       cl::Required, cl::cat(ToolOptions));
+static cl::list<std::string>
+    Names("names", cl::ZeroOrMore, cl::CommaSeparated,
+          cl::desc("<Compilation member(s) to be displayed>"),
+          cl::cat(ToolOptions));
+static cl::opt<bool> DigestOnly("digest-only", cl::init(false),
+                                cl::desc("Display the digest only"),
+                                cl::cat(ToolOptions));
+static cl::opt<bool>
+    UseComma("comma", cl::init(false),
+             cl::desc("Output fields are comma (rather than CR) separated"),
+             cl::cat(ToolOptions));
+static cl::opt<bool> Verbose("verbose", cl::init(false),
+                             cl::desc("Produce verbose output"),
+                             cl::cat(ToolOptions));
+
+} // anonymous namespace
+
+raw_ostream &operator<<(raw_ostream &OS, pstore::index::digest const &Digest) {
+  return OS << Digest.to_hex_string();
+}
+
+static std::string getRepoPath() {
+  if (RepoPath.getNumOccurrences() == 0) {
+    // TODO: remove this envrionment variable once the matching behavior is
+    // removed from the compiler.
+    if (auto File = getenv("REPOFILE")) {
+      return {File};
+    }
+  }
+  return RepoPath;
+}
+
+int main(int argc, char *argv[]) {
+  cl::HideUnrelatedOptions({&ToolOptions});
+  cl::ParseCommandLineOptions(argc, argv,
+                              "A repo utility for dumping the names and "
+                              "digests using a ticket file.\n");
```

Comment:
I think we should probably call it “program repository” rather than “repo” here for the benefit of those unfamiliar with our short-form name. How about ”A utility to dump a compilation's names and digests from a program repository”. The ticket-file stuff will come from the required TicketPath option.

Reply:
Thanks, changed it.